openapi/hooks.py

```diff
@@ -114,3 +117,100 @@ def exclude_paths_hook(endpoints, **kwargs):  # noqa: ARG001
         for (path, path_regex, method, callback) in endpoints
         if not any(path.startswith(prefix) for prefix in EXCLUDED_PATHS)
     ]
+
+
+def insert_wagtail_pages_schema(endpoints):
+    """
+    Insert Wagtail pages schema into the OpenAPI endpoints.
+    """
+
+    from cms.wagtail_api.schema.views import (
+        WagtailCertificatePagesRetrieveSchemaView,
+        WagtailCertificatePagesSchemaView,
+        WagtailCoursePagesSchemaView,
+        WagtailPagesRetrieveSchemaView,
+        WagtailPagesSchemaView,
+        WagtailProgramPagesSchemaView,
+    )
+
+    class WrappedPagesView(WagtailPagesSchemaView):
+        pass
+
+    class WrappedCertificateView(WagtailCertificatePagesSchemaView):
+        pass
+
+    class WrappedCourseView(WagtailCoursePagesSchemaView):
+        pass
+
+    class WrappedProgramView(WagtailProgramPagesSchemaView):
+        pass
+
+    class WrappedPagesRetrieveView(WagtailPagesRetrieveSchemaView):
+        pass
+
+    class WrappedCertificateRetrieveView(WagtailCertificatePagesRetrieveSchemaView):
+        pass
+
+    pages_view = WrappedPagesView.as_view()
+    certificate_view = WrappedCertificateView.as_view()
+    course_view = WrappedCourseView.as_view()
+    program_view = WrappedProgramView.as_view()
+    retrieve_view = WrappedPagesRetrieveView.as_view()
+    certificate_retrieve_view = WrappedCertificateRetrieveView.as_view()
+
+    # manually attach the class for schema inspection
+    pages_view.cls = WrappedPagesView
+    certificate_view.cls = WrappedCertificateView
+    course_view.cls = WrappedCourseView
+    program_view.cls = WrappedProgramView
+    retrieve_view.cls = WrappedPagesRetrieveView
+    certificate_retrieve_view.cls = WrappedCertificateRetrieveView
+    endpoints.append(
+        (
+            "/api/v2/pages/",
+            "^api/v2/pages/$",
+            "GET",
+            pages_view,
+        )
+    )
+    endpoints.append(
+        (
+            "/api/v2/pages/?fields=*&type=cms.CertificatePage",
+            "^api/v2/pages/?fields=*&type=cms.CertificatePage$",
+            "GET",
+            certificate_view,
+        )
+    )
+    endpoints.append(
+        (
+            "/api/v2/pages/?fields=*&type=cms.CoursePage",
+            "^api/v2/pages/?fields=*&type=cms.CoursePage$",
+            "GET",
+            course_view,
+        )
+    )
+    endpoints.append(
+        (
+            "/api/v2/pages/?fields=*&type=cms.ProgramPage",
+            "^api/v2/pages/?fields=*&type=cms.ProgramPage$",
+            "GET",
+            program_view,
+        )
+    )
+    endpoints.append(
+        (
+            "/api/v2/pages/<int:pk>/",
```

Comment:
Still issues - I was actually able to get this to work by using `{pk}` for the placeholder and then manually setting the `operation_id` in the `extend_schema` decorator (otherwise it complains about a conflict). Something similar should also work for the certificate retrieve below with the revision ID in it. As is, the client that gets generated ends up with this code in it:

```
const localVarPath = `/api/v2/pages/<int:pk>/`
                .replace(`{${"pk"}}`, encodeURIComponent(String(pk)));
```
which doesn't work (and neither does the Swagger UI). (For what it's worth, spectacular will replace pk with id - not sure why that is.) It might be helpful to grab the mitxonline-api-clients repo and build a client to see what it outputs too.

Reply:
@jkachel Can you take a look at the changes in https://github.com/mitodl/mitxonline/pull/2809/commits/d056a2e389aedf00da31a7d203e4aaee892ee98a?